### ISSUE TITLE
Need a BBE on finite types

### ISSUE THREAD

**Issue by pubudu91** [ballerina-platform/ballerina-distribution#1934]
Currently, there isn't an example on finite types in BBE

**Comment by MaryamZi** [comment]
We have a BBE on type definitions now https://ballerina.io/learn/by-example/type-definitions.

Also since the spec doesn't explicitly mention "finite" types, do we still need a separate BBE for this?

@pubudu91, thoughts please?

**Comment by pubudu91** [comment]
Yeah, don't need a separate BBE for this. But the new union example isn't illustrative enough to cover this case. Unsure whether a BBE is the place for it or whether a more comprehensive language guide would be more suitable for such details.
